File: 2785-sort-vowels-in-a-string/2785-sort-vowels-in-a-string.java
class Solution {
    public String sortVowels(String s) {

        List<Character> list = new ArrayList<>();
        for (char c : s.toCharArray()) 
            if ("AEIOUaeiou".indexOf(c) != -1) list.add(c);

        Collections.sort(list);

        int idx = 0;
        StringBuilder sb = new StringBuilder();
        
        for (char c : s.toCharArray()) 
            sb.append("AEIOUaeiou".indexOf(c) != -1 ? list.get(idx++) : c);
        
        return sb.toString();
    }
}

File: 2785-sort-vowels-in-a-string/2785-sort-vowels-in-a-string.js
var sortVowels = function(s) {
    const arr = new Array();

    for (const c of s) 
        if ("AEIOUaeiou".includes(c)) arr.push(c);

    arr.sort();

    let i = 0;
    let sb = "";

    for (const c of s) 
        sb += "AEIOUaeiou".includes(c) ? arr[i++] : c;
    
    return sb;
};